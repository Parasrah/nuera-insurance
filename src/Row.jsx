import React from 'react'
import { StyleSheet, css } from 'aphrodite'
import PropTypes from 'prop-types'
import { toast } from 'react-toastify'

import globalStyles from '@/styles'
import Trash from '@/icons/trash-2.svg'

function Row ({ name, value, id, deleteRow, colored }) {
  return (
    <div className={css(globalStyles.rowContainer, styles.row, colored && styles.colored)}>
      <span className={css(styles.grow, styles.name)}>{name}</span>
      <span className={css(styles.grow, styles.value)}>${value}</span>
      <Trash
        className={css(styles.trash)}
        onClick={() => deleteRow(id).match({
          onErr (err) {
            toast.error(`Error: ${err.message}`)
          },
          onOk () {}
        })}
      />
    </div>
  )
}

const styles = StyleSheet.create({
  row: {
    color: 'rgba(0, 0, 0, 0.8)',
    transition: 'font-size 0.3s ease',
    ':hover': {
      fontSize: '1.08rem'
    },
    fontWeight: 300
  },
  trash: {
    cursor: 'pointer'
  },
  grow: {
    flex: '1 0 0'
  },
  name: {
  },
  value: {
    paddingLeft: '10px'
  },
  colored: {
    backgroundColor: 'rgba(255, 255, 255, 1)'
  }
})

Row.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  deleteRow: PropTypes.func.isRequired,
  colored: PropTypes.bool.isRequired
}

export { Row }
