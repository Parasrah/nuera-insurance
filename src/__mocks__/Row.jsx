import React from 'react'

function Row ({ name, value, id }) {
  return <div className={`${name} ${value} ${id}`} />
}

export { Row }
