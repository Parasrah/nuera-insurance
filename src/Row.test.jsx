/* globals test, describe, expect */
import { Row } from '@/Row'
import React from 'react'
import renderer from 'react-test-renderer'

describe('#render', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Row
        name='TV'
        value={979.99}
        category='Electronics'
        id={6}
        deleteRow={() => {}}
        colored={true}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
