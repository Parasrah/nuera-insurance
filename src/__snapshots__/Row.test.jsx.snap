// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders correctly 1`] = `
<div
  className="rowContainer row colored"
>
  <span
    className="grow name"
  >
    TV
  </span>
  <span
    className="grow value"
  >
    $
    979.99
  </span>
  <test-file-stub
    className="trash"
    onClick={[Function]}
  />
</div>
`;
